### PR TITLE
CLN: Using sum instead of looping

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -120,13 +120,9 @@ def memory_usage_of_objects(arr: object[:]) -> int64_t:
...
     i: Py_ssize_t
     n: Py_ssize_t
-    size: int64_t
 
-    size = 0
     n = len(arr)
-    for i in range(n):
-        size += arr[i].__sizeof__()
-    return size
+    return sum(arr[i].__sizeof__() for i in range(n))
```
Comment (jbrockmendel) [inline pandas/_libs/lib.pyx]: have you profiled this? i expect this is slower

Comment (jbrockmendel) [reply pandas/_libs/lib.pyx]: (and if im wrong, then how about going one step further and just do `sum(x.__sizeof__() for x in arr)`?)

Comment (MomIsBestFriend) [reply pandas/_libs/lib.pyx]: I have not profiled this, as ```asv``` requires conda to be installed, and I use ```venv```.

**Comment by WillAyd** [comment]
@momisbestfriend just as a general comment Python idioms may not always be applicable for Cython. 

Changes like this definitely need benchmarks to go with them

File: pandas/_libs/lib.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/lib.pyx]: 1) `%timeit` profiling is a good alternative, especially for small things like this
2) i use `asv continuous -E virtualenv`

**Comment by MomIsBestFriend** [comment]
I bench marked it (hopefully I did it well)

The test I use to check both is the following:

```python
import numpy as np

from pandas._libs.lib import memory_usage_of_objects

arr = np.random.rand(1, 100)

%timeit memory_usage_of_objects(arr)
```

---

master:
```
In [4]: %timeit memory_usage_of_objects(arr)                                                                  
268 ns ± 16.9 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

this PR:
```
In [4]: %timeit memory_usage_of_objects(arr)                                                                  
463 ns ± 9.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

**Comment by WillAyd** [comment]
Yea that's a pretty large slowdown. Still ns which is good and I doubt anyone is calling this function in a tight loop so probably invisible to the end user, but in any case I don't think worth changing.

Thanks in any case!

**Pull request closed without merging**